tools/mo/openvino/tools/mo/utils/cli_parser.py

```diff
@@ -6,21 +6,608 @@
+            return shape_to_str(input_shape, ',')
+        # The case when list of shapes is specified
+        shapes = []
+        for shape in input_shape:
+            shapes.append(shape_to_str(shape, ','))
+        return ','.join(shapes)
+    return shape_to_str(input_shape, ',')
+
+
+def type_to_str(type_obj):
+    if isinstance(type_obj, str):
+        return type_obj
+    if isinstance(type_obj, type):
+        return np_data_type_to_destination_type(type_obj)
+    if isinstance(type_obj, Type):
+        return type_obj.get_type_name()
+    raise Exception("Incorrect type. Expected Type or numpy type, got {}".format(type(type_obj)))
+
+
+def value_to_str(value, separator):
+    if isinstance(value, np.ndarray):
+        values = []
+        for x in np.nditer(value):
+            values.append(str(x))
+        return "[" + separator.join(values) + "]"
+    if isinstance(value, list):
+        values = []
+        for x in value:
+            values.append(str(x))
+        return "[" + separator.join(values) + "]"
+    raise Exception("Incorrect value type. Expected np.ndarray or list, got {}".format(type(value)))
+
+
+def single_input_to_str(input):
+    if isinstance(input, str):
+        return input
+    if isinstance(input, openvino.tools.mo.InputCutInfo):
+        if not isinstance(input.name, str):
+            raise Exception("Input name should be string, got {}".format(input.name))
+        input_str = input.name
+        assert input_str is not None, "Incorrect InputCutInfo. 'name' should be set."
+        if input.shape is not None:
+            input_str += shape_to_str(input.shape, " ")
+        if input.type is not None:
+            input_str += "{" + type_to_str(input.type) + "}"
+        if input.value is not None:
+            input_str += "->" + value_to_str(input.value, " ")
+        return input_str
+
+
+def input_to_str(input):
+    if input is None or isinstance(input, str):
+        return input
+    if isinstance(input, list):
+        inputs_str = []
+        for inp in input:
+            inputs_str.append(single_input_to_str(inp))
+        return ','.join(inputs_str)
+    return single_input_to_str(input)
+
+
+def mean_scale_value_to_str(value):
+    # default empty value
+    if isinstance(value, tuple) and len(value) == 0:
+        return value
+
+    if isinstance(value, str):
+        return value
+    if isinstance(value, dict):
+        values_str = []
+        for op_name, val in value.items():
+            if not isinstance(op_name, str):
+                raise Exception("Incorrect operation name type. Expected string, got {}".format(type(op_name)))
+            values_str.append(op_name + value_to_str(val, ","))
+        return ",".join(values_str)
+    return value_to_str(value, ",")
+
+
+def layout_to_str(layout):
+    if isinstance(layout, str):
+        return layout
+    if isinstance(layout, Layout):
+        return layout.to_string()
```

Comment:
PartialShape to string methods are located in IR fronted, so it is not convenient to use them. And there is no Python bindings for them. It will be fixed in this ticket: 89901